src/main/scala/fi/oph/kouta/service/ToteutusService.scala

```diff
@@ -38,6 +38,12 @@ abstract class ToteutusService(sqsInTransactionService: SqsInTransactionService)
   def listHakukohteet(oid: ToteutusOid)(implicit authenticated: Authenticated): Seq[HakukohdeListItem] =
     withRootAccess(Role.Hakukohde.readRoles)(HakukohdeDAO.listByToteutusOid(oid))
 
+  def listHakukohteet(oid: ToteutusOid, organisaatioOid: OrganisaatioOid)(implicit authenticated: Authenticated): Seq[HakukohdeListItem] = {
+    withAuthorizedChildOrganizationOids(organisaatioOid, Role.Toteutus.readRoles) {
```

Comment:
En ole ihan varma, pitäisikö tässä olla toteutuksen vai hakukohteen readRoles. Oikeasti pitäisi olla varmasti molemmat. Vaihtaisin ehkä kuitenkin tähän nyt sen Role.Hakukohde.readRoles. Voin vielä viilata näitä oikeuksia tuolla toisessa branhissa. Pitänee tehdä mahdolliseksi tarkistaa kahdet eri roolit.